internal/uploader.go

```diff
@@ -216,3 +227,32 @@ func (uploader *SplitStreamUploader) Clone() Uploader {
 		blockSize:  uploader.blockSize,
 	}
 }
+
+func (uploader *RegularUploader) ShowRemainingTime() {
+	defer uploader.waitGroup.Done()
+	startTime := time.Now()
+	for {
+		compressedSizeInt, err := uploader.UploadedDataSize()
+		if err != nil {
+			return
+		}
+		uncompressedSizeInt, err := uploader.RawDataSize()
+		if err != nil {
+			return
+		}
+
+		compressedSize := float64(compressedSizeInt) / 1000000
```

Comment:
Let's adopt some code to get a nice output for units and place it in the `util` package:
```
func ByteCountSI(b int64) string {
	const unit = 1000
	if b < unit {
		return fmt.Sprintf("%d B", b)
	}
	div, exp := int64(unit), 0
	for n := b / unit; n >= unit; n /= unit {
		div *= unit
		exp++
	}
	return fmt.Sprintf("%.1f %cB",
		float64(b)/float64(div), "kMGTPE"[exp])
}
```